VMBackup/main/safefreeze/src/safefreeze.c

```diff
@@ -16,19 +16,36 @@
 
 
 #include <stdio.h>
+#include <stdarg.h>
 #include <stdlib.h>
 #include <fcntl.h>
 #include <signal.h>
 #include <linux/fs.h>
 #include <sys/ioctl.h>
+#include <time.h>
+#include <string.h>
 
 
 #define JUMPWITHSTATUS(x)        \
 {                                \
     status = (x);                \
     if (status) goto CLEANUP;    \
 }
-
+void logger(const char *logstr,...)
+{
+    time_t mytime;
+    struct tm * timeinfo;
+    char buffer [80];
+    time(&mytime);
+    timeinfo = localtime(&mytime);
+    strftime (buffer,80,"%F %X",timeinfo);
```

Comment:
Use uniform spacing, formatting and indentations for code.

function<nospace>(param1,<space>param2,...);